### PR TITLE
Liqo installer improvement for Liqo Agent installation

### PR DESCRIPTION
**Pull request by AbakusW**
# Description

This PR adds a section to the Liqo installer to let users install and uninstall the latest version of the Liqo Agent as a desktop application.

- **[ADDED]** files to make the Agent binary as a desktop application
  - xdg *.desktop* file
  - *.svg* scalable logo

- **[ADDED]** new iconset for the desktop banner notifications

- **[ADDED]** *setup_installation_platform()* function in the installer that enables to check the OS platform the script is running on, thus starting to address #352 issue on OS compatibility. Currently only the *Liqo Agent* installer exploits this mechanism, avoiding its binary installation on platforms differing from Linux-based ones.

### CODE REVIEW

**Review by giorio94 — CHANGES_REQUESTED** [review]
First, thanks @AbakusW for having split down this PR. Much easier to read and review now.

I left some comments inline. Most simply relate to style and should be very easy to address. 
At the core, to me, the `get_liqo_agent` function should be improved, especially for what regards getting the correct release instead of using always the latest one.

Additionally, the `get_agent_link` and `setup_agent_environment` should be good candidates for a couple of unit tests if you want to try implementing them.

Finally, I feel a bit odd having icon files inside the `scripts` directory. Shouldn't them be in an appropriate directory?

As always, feel free to ask if something is not clear or a comment seems wrong. Definitely possible.

File: install.sh [hunk 1]
```diff
@@ -198,6 +201,45 @@ function get_repo_master_commit() {
...
+		AGENT_ASSET_URL=""
+	else
+		AGENT_ASSET_URL=$(download ${latest_url} | grep -Po '"browser_download_url": "\K.*liqo-agent.*(?=")' || echo "")
+	fi
+}
+
+function setup_agent_environment() {
+	[[ -z ${HOME:-} ]] && HOME=~
+	# Default directory for XDG_CONFIG_HOME.
+	AGENT_XDG_CONFIG_DIR="${HOME}/.config/"
```
[the remark this answers is not in the document]
Comment (AbakusW) [reply install.sh:222]: thanks for this!

File: install.sh [hunk 2]
```diff
@@ -198,6 +201,45 @@ function get_repo_master_commit() {
...
+	# Liqo subdirectory containing the notifications icons.
+	AGENT_ICONS_DIR="${AGENT_XDG_DATA_DIR}/liqo/icons"
+	# Directory storing the '.desktop' file.
+	AGENT_APP_DIR="${AGENT_XDG_DATA_DIR}/applications"
+	# Directory storing the scalable icon for the desktop application.
+	AGENT_THEME_DIR="${AGENT_XDG_DATA_DIR}/icons/hicolor/scalable/apps"
+	# Directory storing the '.desktop' file to enable the application autostart.
+	AGENT_AUTOSTART_DIR="${AGENT_XDG_CONFIG_DIR}/autostart"
+
+	if [[ $# -ge 1 ]] && [[ "$1" == "-c" ]]; then
```
[the remark this answers is not in the document]
Comment (AbakusW) [reply install.sh:237]: As for similar changes, thank you for having me discover this quick fallback bash syntax I never heard of

**Review by giorio94 — COMMENTED** [review]
Thanks @AbakusW for all the improvements.

Just a couple of additional suggestions. Very easy to address (most are just stylistic), but some are definitely necessary as the installer currently fails to install the agent (at least when I tried). Please, after fixing them, double check that the installation process completes correctly.

A second problem is that the installation of the v0.1 is broken, as the layout of the repo changed (it fails to retrieve the icons). My suggestion there is to manually delete the binary from that release (which I nonetheless suppose not being yet fully functional), and do a follow-up release when everything will be ready. @palexster WDYT?

**Comment by palexster** [comment]
> A second problem is that the installation of the v0.1 is broken, as the layout of the repo changed (it fails to retrieve the icons). My suggestion there is to manually delete the binary from that release (which I nonetheless suppose not being yet fully functional), and do a follow-up release when everything will be ready. @palexster WDYT?

Ok to me. We will publish the agent for the v0.2 or make a v0.1.1 with the agent correctly generated.

**Comment by AbakusW** [comment]
> 
> 
> > A second problem is that the installation of the v0.1 is broken, as the layout of the repo changed (it fails to retrieve the icons). My suggestion there is to manually delete the binary from that release (which I nonetheless suppose not being yet fully functional), and do a follow-up release when everything will be ready. @palexster WDYT?
> 
> Ok to me. We will publish the agent for the v0.2 or make a v0.1.1 with the agent correctly generated.

@palexster seems good to me too, this way we can also have the opportunity to merge #214 and #351 first, thus having a fully functioning version with a couple of working features

**Review by palexster — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]
/lgtm

Two minor things I noticed just now, the icon and the desktop file should probably be moved to the assets directory, to leave only proper scripts in the scripts directory

**Pull request merged**